pkg/controller/hibernation/hibernation_controller.go

```diff
@@ -212,6 +212,26 @@ func (r *hibernationReconciler) Reconcile(ctx context.Context, request reconcile
 		return reconcile.Result{}, r.updateClusterDeploymentStatus(cd, cdLog)
 	}
 
+	// Secret toggle allowing consumers to e.g. perform manual maintenance on machines without the
+	// controller attempting to reconcile desired CD.Spec.PowerState.
+	if paused, err := strconv.ParseBool(cd.Annotations[constants.PowerStatePauseAnnotation]); err == nil && paused {
```

Comment:
Is someone had already applied this annotation but subsequently removes it, as per current logic, hibernation controller would attempt to run the cluster - I have not fully thought out the scenario, but we should have a way of either working with Unknown powerstate and powerstate paused condition reasons, or give an error early

Reply:
Yup, I ran through all off that, which is how I came up with the second commit. Again, it did work, but the timing and transitions looked odd. With the second commit it’s smoother.